src/graphene-rect.c

```diff
@@ -829,3 +829,36 @@ graphene_rect_zero (void)
 {
   return &_graphene_rect_zero;
 }
+
+/**
+ * graphene_rect_scale:
+ * @s_h: horizontal scale factor
+ * @s_v: vertical scale factor
+ * @res: (out caller-allocates): return location for the
+ *   scaled rectangle
+ *
+ * Scales the size and origin of a rectangle horizontaly by @s_h,
+ * and vertically by @s_v. The result @res is normalized.
+ *
+ * Returns: (transfer none): a scaled rectangle
+ *
+ * Since: 1.8.6
+ */
+graphene_rect_t *
+graphene_rect_scale (const graphene_rect_t *r,
+                     float                  s_h,
+                     float                  s_v,
+                     graphene_rect_t       *res)
+{
+  if (res != r)
+    *res = *r;
```

Comment:
You should use `graphene_rect_normalize_r (r, res)` and then modify `res`.

Reply:
Done